### PR TITLE
Turn on exact_dtype by default on test_sparse.py (#34489)

### PR DESCRIPTION
**Pull request by terhuhf**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34542 Turn on exact_dtype by default on test_sparse.py (#34489)**

Turn on exact_dtype by default on test_sparse.py (#34489)

Summary:
Pull Request resolved: #34489

Test Plan:

```
python test/test_sparse.py
```

Reviewers:

Subscribers:

Tasks:
https://github.com/pytorch/pytorch/issues/34489

Tags:

Differential Revision: [D20369764](https://our.internmc.facebook.com/intern/diff/D20369764)

### CODE REVIEW

File: test/test_sparse.py [hunk 1]
```diff
@@ -227,8 +228,11 @@ def test_tensor(x, res):
             x.to_dense()  # Tests triple to_dense for memory corruption
             x.to_dense()
             x.to_dense()
-            self.assertEqual(res, x.to_dense())
-            self.assertEqual(res, self.safeToDense(x))
+            # We dont have to_dense for half types, so we don't request 
+            # exact_dtype when testing. We allow to_dense to convert dtype
+            # to torch.float64
```
Comment (terhuhf) [inline test/test_sparse.py]: What is the semantics of to_dense working with torch.float32?

File: test/test_sparse.py [hunk 2]
```diff
@@ -227,8 +228,11 @@ def test_tensor(x, res):
             x.to_dense()  # Tests triple to_dense for memory corruption
             x.to_dense()
             x.to_dense()
-            self.assertEqual(res, x.to_dense())
-            self.assertEqual(res, self.safeToDense(x))
+            # We dont have to_dense for half types, so we don't request
```
Comment (albanD) [inline test/test_sparse.py]: if the problem is only with half, why not use `exact_dtype=x.dtype is not torch.float16` ?

File: test/test_sparse.py (hunk 1 above)
Comment (ezyang) [reply test/test_sparse.py]: Should work just fine.

File: test/test_sparse.py [hunk 3]
```diff
@@ -243,6 +247,7 @@ def fn(x):
         # we don't have to_dense for half types on CPU because it is implemented
         # with a slower add_ operation
         for dtype in [torch.float16, torch.float64] if self.device != 'cpu' else [torch.float64]:
+
```
Comment (ezyang) [inline test/test_sparse.py]: nit: Avoid extraneous whitespace changes when working on diffs! They make reviewers unhappy ;)

**Review by ezyang — APPROVED** [review]
Looks great!

**Pull request closed without merging**